src/index.ts

```diff
@@ -209,4 +211,77 @@ export class SilentPayment {
   static addressToPubkey(address: string): string {
     return uint8ArrayToHex(bitcoin.address.toOutputScript(address).subarray(2));
   }
+
+  static getPubkeysFromTransactionInputs(tx: Transaction): Uint8Array[] {
+    const result: Uint8Array[] = [];
+
+    const stackToPubkeys = (stack: Stack): Uint8Array[] => {
+      return stack
+        .filter((elem) => typeof elem !== "number") // filtering out numbers, leaving only Uint8Array
+        .filter((elem) => ecc.isXOnlyPoint(elem as Uint8Array)) as Uint8Array[];
```

Comment:
@junderw in your origina implementationit had `isCanonicalPubKey`, but that didnt fly, so i switched to this "x-only pubkey". is there an equivalent in bitcoinjslib for that..?

Reply:
Nope. `ecc.isXOnlyPoint` is really the only sane way to do it... since x-only style pubkeys just look like a 32 byte blob, could be a SHA256 hash for all we know... in fact, there are more pubkeys than non-pubkeys in the 32 byte possibility space, so it's highly possible that `ecc.isXOnlyPoint` will return true even though the creator of the script did not intend for it to be used as a pubkey.

Detecting pubkeys in a taproot script is very hard.

`ecc.isXOnlyPoint` will at least guarantee that the ones that return true are a valid X coordinate on the curve (ie. a valid pubkey)

Reply:
changed to:

`.filter((elem) => ecc.isXOnlyPoint(elem as Uint8Array) || script.isCanonicalPubKey(elem as Uint8Array)) as Uint8Array[];`